src/Accelerators/NNPA/Conversion/ONNXToZHigh/ONNXLegalityCheck.cpp

```diff
@@ -231,41 +267,98 @@ StringRef getStrPaddingType(OP op) {
+    }
     // Output tensor's height and width dimensions must be 1.
-    if (outputShape != 1)
-      return false;
+    if (outputShape != 1) {
+      std::string message = "When the strides is zero, output tensor's height "
+                            "and width dimensions (" +
+                            std::to_string(outputShape) + ") must be 1.";
+      return emitWarningMessageNNPAUnsupported(op, StringRef(message));
+    }
     // padding_type must be VALID_PADDING.
-    if (!paddingType.equals("VALID_PADDING"))
-      return false;
+    if (!paddingType.equals("VALID_PADDING")) {
+      std::string message = "When the strides is zero, padding type (" +
+                            paddingType.str() + ") must be VALID_PADDING.";
+      return emitWarningMessageNNPAUnsupported(op, StringRef(message));
+    }
   } else {
     // strides are greater than zero
     // kernel_width and kernel_height must be less than or equal to 64.
-    if (kernelShape > 64)
-      return false;
+    if (kernelShape > 64) {
+      std::string message = "When the strides are greater than zero, the "
+                            "kernel_width and kernel_height (" +
+                            std::to_string(kernelShape) +
+                            ") must be less than or equal to 64.";
+      return emitWarningMessageNNPAUnsupported(op, StringRef(message));
+    }
     if (paddingType.equals("SAME_PADDING")) {
-      if (outputShape != ceil((float)inputShape / strides))
-        return false;
+      int64_t reqOutputShape = ceil((float)inputShape / strides);
+      if (outputShape != reqOutputShape) {
+        std::string message =
+            "When the padding type is SAME_PADDING, output tensor's height and "
+            "width dimensions (" +
+            std::to_string(outputShape) +
+            ") must be equal with ceil((float)inputShape / strides)) (=" +
+            std::to_string(reqOutputShape) + ").";
+        return emitWarningMessageNNPAUnsupported(op, StringRef(message));
+      }
     } else { // VALID_PADDING
-      if (outputShape != ceil((float)(inputShape - kernelShape + 1) / strides))
-        return false;
+      int64_t reqOutputShape =
+          ceil((float)(inputShape - kernelShape + 1) / strides);
+      if (outputShape != reqOutputShape) {
+        std::string message = "When the padding type is VALID_PADDING, output "
+                              "tensor's height and width dimensions (" +
+                              std::to_string(outputShape) +
+                              ") must be equal with ceil((float)(inputShape - "
+                              "kernelShape + 1) / strides)) (= " +
+                              std::to_string(reqOutputShape) + ").";
+        return emitWarningMessageNNPAUnsupported(op, StringRef(message));
+      }
     }
   }
   return true;
 }
 
+bool emitWarningMessageNNPAUnsupported(Operation *op, StringRef message) {
```

Comment:
Do we really need to use StringRef instead of just `const std::string&`? I don't see any operation on `message` and passing a pointer looks faster than constructing a StringRef.

Reply:
Updated. Thanks!